pkg/kubelet/kubelet_pods.go

```diff
@@ -381,6 +383,7 @@ func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, h
 			Name:              mount.Name,
 			ContainerPath:     containerPath,
 			HostPath:          hostPath,
+			Image:             image,
```

Comment:
I wonder how it worked without the image before? What is the image field is being used for if it was not passed here before and nothing broke?

Reply:
It aims to add all missing mount information to the runtime and let them decide when the volume source is image.

Reply:
@SergeyKanzhelev  some volume attributes reported by the `image` volume plugin, i.e. relabel, read-only, etc., aren't sent to the implementation of CRI runtime in [previous kubelet implementation](https://github.com/kubernetes/kubernetes/pull/125663).  

```go
func makeMounts(pod *v1.Pod, podDir string, container *v1.Container, hostName, hostDomain string, podIPs []string, podVolumes kubecontainer.VolumeMap, hu hostutil.HostUtils, subpather subpath.Interface, expandEnvs []kubecontainer.EnvVar, supportsRRO bool, imageVolumes kubecontainer.ImageVolumes) ([]kubecontainer.Mount, func(), error) {
	mountEtcHostsFile := shouldMountHostsFile(pod, podIPs)
	klog.V(3).InfoS("Creating hosts mount for container", "pod", klog.KObj(pod), "containerName", container.Name, "podIPs", podIPs, "path", mountEtcHostsFile)
	mounts := []kubecontainer.Mount{}
	var cleanupAction func()
	for i, mount := range container.VolumeMounts {
		// Check if the mount is referencing an OCI volume
		if imageVolumes != nil && utilfeature.DefaultFeatureGate.Enabled(features.ImageVolume) {
			if image, ok := imageVolumes[mount.Name]; ok {
				mounts = append(mounts, kubecontainer.Mount{
					Name:          mount.Name,
					ContainerPath: mount.MountPath,
					Image:         image,
				})
				continue
			}
		}
                 ...skipped..
		mounts = append(mounts, kubecontainer.Mount{
			Name:              mount.Name,
			ContainerPath:     containerPath,
			HostPath:          hostPath,
			ReadOnly:          mount.ReadOnly || mustMountRO,
			RecursiveReadOnly: rro,
			SELinuxRelabel:    relabelVolume,
			Propagation:       propagation,
		})
	}
                 ...skipped..
	return mounts, cleanupAction, nil
```

What's the result?  Please see the `without this fix` section in my PR's description.